sentry/src/main/java/io/sentry/DisabledQueue.java

```diff
@@ -0,0 +1,115 @@
+package io.sentry;
+
+import java.io.Serializable;
+import java.util.AbstractCollection;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Queue;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+final class DisabledQueue<E> extends AbstractCollection<E> implements Queue<E>, Serializable {
+
+  /** Serialization version. */
+  private static final long serialVersionUID = -8423413834657610417L;
+
+  /** Constructor that creates a queue that does not accept any element. */
+  public DisabledQueue() {}
+
+  // -----------------------------------------------------------------------
+  /**
+   * Returns the number of elements stored in the queue.
+   *
+   * @return this queue's size
+   */
+  @Override
+  public int size() {
+    return 0;
+  }
+
+  /**
+   * Returns true if this queue is empty; false otherwise.
+   *
+   * @return false
+   */
+  @Override
+  public boolean isEmpty() {
+    return false;
```

Comment:
@lucas-zimerman @stefanosiano should this return `true` instead? If no, can we document why?